scripts/us_fema/national_risk_index/generate_schema_and_tmcf.py

```diff
@@ -0,0 +1,520 @@
+# measures, we consider it as different units measuring the same
+# StatisticalVariable
+import json
+import pandas as pd
+import numpy as np
+import logging
+
+IGNORED_FIELDS = [
+    "OBJECTID", "Shape", "Shape_Length", "Shape_Area", "STATE", "STATEABBRV",
+    "STATEFIPS", "COUNTY", "COUNTYTYPE", "COUNTYFIPS", "STCOFIPS", "NRI_ID",
+    "TRACT", "TRACTFIPS", "POPULATION", "BUILDVALUE", "AGRIVALUE", "AREA",
+    "NRI_VER", "AIANNHCE", "FEDREG2020", "FEDERAL_ID", "JURS_NAME", "JURS_AREA",
+    "JURS_TYPE", "HIFLD_NAME", "HIFLD_AREA", "HIFLD_TYPE"
+]
+
+# i/o filenames
+NRI_DATADICTIONARY_INFILE_FILENAME = "source_data/NRIDataDictionary.csv"
+SCHEMA_OUTFILE_FILENAME = "output/fema_nri_stat_vars.mcf"
+TMCF_OUTFILE_FILENAME = "output/fema_nri_counties.tmcf"
+CSV_COLUMNS_FILENAME = "output/csv_columns.json"
+
+# feature flags
+# skip {Annualized Frequency, Historic Loss Ratio, Exposure}
+FLAG_SKIP_EAL_COMPONENTS = True
+# skip {Building, Population, Population Equivalence, Agriculture}
+FLAG_SKIP_IMPACTED_THING_COMPONENTS = True
+# skip {Rating, National Percentile, State Percentile}
+FLAG_SKIP_NON_SCORE_RELATIVE_MEASURES = True
+# skip {Community Resilience Raw Value, Social Vulnerability Raw Value}
+FLAG_SKIP_RAW_VALUES = True
+
+# hard coded lists of interest
+COMPOSITE_ROW_LAYERS = [
+    "National Risk Index", "Expected Annual Loss", "Social Vulnerability",
+    "Community Resilience"
+]
+EAL_COMPONENTS = [
+    "Number of Events", "Annualized Frequency", "Historic Loss Ratio",
+    "Exposure"
+]
+IMPACTED_THING_COMPONENTS = ["Building", "Population", "Agriculture"]
+RAW_VALUE_COMPONENTS = [
+    "Social Vulnerability - Value", "Community Resilience - Value"
+]
+NON_SCORE_RELATIVE_MEASURES = ["Rating", "Percentile"]
+
+# template strings
+COMPOSITE_MCF_FORMAT = """Node: dcid:{node_dcid}
+typeOf: dcs:StatisticalVariable
+statType: dcs:measuredValue
+populationType: dcs:NaturalHazardImpact
+measuredProperty: dcs:{m_prop}
+name: "{statvar_name}"
+"""
+HAZARD_MCF_FORMAT_BASE = """Node: dcid:{node_dcid}
+typeOf: dcs:StatisticalVariable
+populationType: dcs:NaturalHazardImpact
+statType: dcs:measuredValue
+naturalHazardType: dcs:{haz_type}
+measuredProperty: dcs:{m_prop}
+name: "{statvar_name}"
+"""
+TMCF_FORMAT = """
+Node: E:FEMA_NRI->E{index}
+typeOf: dcs:StatVarObservation
+variableMeasured: dcs:{statvar_dcid}
+observationAbout: C:FEMA_NRI_Counties->DCID_GeoID
+observationDate: "{obs_date}"
+value: C:FEMA_NRI_Counties->{field_name} 
+"""
+
+SCHEMA_OUTPUT_PREFIX = """
+# This file was autogenerated. Please see:
+# https://github.com/datacommonsorg/data/tree/master/scripts/fema/nri/README.md
+
+"""
+
+# mappings
+DATACOMMONS_ALIASES = {
+    "Score": "FemaNationalRiskScore",
+    "SocialVulnerability": "femaSocialVulnerability",
+    "CommunityResilience": "femaCommunityResilience",
+    "HazardTypeRiskIndex": "femaNaturalHazardRiskIndex",
+    "Hazard Type Risk Index": "femaNaturalHazardRiskIndex",
```

Comment:
In what cases do we have a space vs. not?  "HazardTypeRiskIndex" vs. "Hazard Type Risk Index"

An alternative is to canonicalize (strip space and lower-case), the key in the map.